frontend/src/components/widgets/NumberInput/NumberInput.tsx

```diff
@@ -204,6 +204,22 @@ class NumberInput extends React.PureComponent<Props, State> {
     if (this.state.dirty) {
       this.commitWidgetValue({ fromUi: true })
     }
+
+    const node = this.inputRef.current
+    const nodeContainer = node?.parentElement?.parentElement?.parentElement
+
+    if (nodeContainer) {
+      nodeContainer.classList.remove("focused")
+    }
+  }
```

Comment:
The more Reacty way to do this would be: 

```jsx
private onBlur = (): void => {
  ...
  this.setState({ isFocused: false })
}

private onFocus = (): void => {
  ...
  this.setState({ isFocused: true })
}

public render(...) {
   ...
   return (
    <div className="stNumberInput" style={style} className={isFocused ? 'foo' : 'bar'}>
    ...
   )
}
```

Reply:
Makes sense! Updated in [3ed1629](https://github.com/streamlit/streamlit/pull/5193/commits/3ed162951bc50e0759ebbbbdd5b276480f73a250)